### PR TITLE
Parse empty strings as nullable Guid

### PR DESCRIPTION
**Pull request by AntonioSCoelho**
Related to [msgraph-sdk-dotnet/issues/2447](https://github.com/microsoftgraph/msgraph-sdk-dotnet/issues/2447)

### CODE REVIEW

**Comment by AntonioSCoelho** [comment]
> @AntonioSCoelho please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree

**Comment by AntonioSCoelho** [comment]
@microsoft-github-policy-service agree


microsoft-github-policy-service[bot] ***@***.***> escreveu
(quarta, 24/04/2024 à(s) 20:03):

File: src/JsonParseNode.cs [hunk 1]
```diff
@@ -125,9 +125,20 @@ public JsonParseNode(JsonElement node, KiotaJsonSerializationContext jsonSeriali
...
+        public Guid? GetGuidValue()
+        {
+            if(_jsonNode.ValueKind != JsonValueKind.String)
+                return null;
+
+            if(_jsonNode.TryGetGuid(out var guid))
+                return guid;
+
+            var guidStr = _jsonNode.GetString();
+            if(string.IsNullOrEmpty(guidStr))
```
Comment (baywet) [inline src/JsonParseNode.cs:137]:
```suggestion
            if(string.IsNullOrEmpty(_jsonNode.GetString()))
```

**Review by baywet — COMMENTED** [review]
Thank you for the contributions!
Can you also:
- add an entry to the changelog (patch, tomorrow's date, changed)
- bump the version in the csproj
Thanks!

**Review by baywet — APPROVED** [review]
Thanks for making the changes!

**Review by andrueastman — COMMENTED** [review]
Thanks for raking the time @AntonioSCoelho

**Pull request merged**